packages/dashboard/src/components/querySettingsSync/utils.ts

```diff
@@ -0,0 +1,17 @@
+type SelectedOption = {
+  label?: string;
+  value?: string;
+};
+
+const SECOND_IN_MS = 1000;
+const MINUTE_IN_MS = 60000;
+export const DEFAULT_REFRESH_RATE = 5000; // 5 seconds
+export const DEFAULT_OPTION = { label: '5s', value: '5000' };
+
+export const refreshRateOptionMap = new Map<number, SelectedOption>([
+  [SECOND_IN_MS, { label: '1s', value: '1000' }],
+  [5 * SECOND_IN_MS, { label: '5s', value: '5000' }],
+  [10 * SECOND_IN_MS, { label: '10s', value: '10000' }],
+  [MINUTE_IN_MS, { label: '1m', value: '60000' }],
+  [5 * MINUTE_IN_MS, { label: '5m', value: '300000' }],
+]);
```

Comment:
I think there's room to improve this handling. Let's talk offline about the following:

```ts
// in types.ts
export const SECOND_IN_MS = 1000;
export const FIVE_SECONDS_IN_MS = 5 * SECOND_IN_MS;
export const TEN_SECONDS_IN_MS = 10 * SECOND_IN_MS;
export const MINUTE_IN_MS = 60 * SECOND_IN_MS;
export const FIVE_MINUTES_IN_MS = 5 * MINUTE_IN_MS;

// use the type everywhere there is a refresh rate (instead of number)
export type RefeshRate = 
  | SECOND_IN_MS 
  | FIVE_SECONDS_IN_MS 
  | TEN_SECONDS_IN_MS 
  | MINUTE_IN_MS 
  | FIVE_MINUTES_IN_MS;

// in utils.ts
import { type SelectProps } from '@cloudscape-design/components';
import { SECOND_IN_MS, FIVE_SECONDS_IN_MS, TEN_SECONDS_IN_MS, MINUTE_IN_MS, FIVE_MINUTES_IN_MS, type RefreshRate } from './types';

interface RefreshRateOption extends SelectProps['options'][number] { 
  value: RefreshRate;
};

const refreshRateOptions = [
  { label: '1s', value: SECOND_IN_MS.toString() },
  { label: '5s', value: FIVE_SECONDS_IN_MS.toString() },
  { label: '10s', value: TEN_SECONDS_IN_MS.toString() },
  { label: '1m', value: MINUTE_IN_MS.toString() },
  { label: '5m', value: FIVE_MINUTES_IN_MS.toString() },
] as const satisfies RefreshRateOption[];

const DEFAULT_REFRESH_RATE_OPTION = refreshRateOptions[1];

const refreshRateOptionMap = refreshRateOptions.reduce((optionMap, { label, value }) => ({ ...optionMap, [parseInt(value, 10)]: { label, value }}), {});

// in component.tsx

// this should work and return a defined value because refresh rate is a discrete set
const currentOption = refreshRateOptionMap[refreshRate];

..

options={refreshRateOptions}
```